api/clients/v2/coretypes/encoded_payload_test.go

```diff
@@ -38,80 +38,20 @@ func TestDecodeLongBytes(t *testing.T) {
-
-		tooLongData := testRandom.Bytes(int(maxPermissiblePayloadLength) + 1)
-		tooLongEncodedPayload := newEncodedPayload(NewPayload(tooLongData))
-		tooLongFieldElements, err := tooLongEncodedPayload.toFieldElements()
-		require.NoError(t, err)
-		// there is one too many field elements for the defined blob length
-		_, err = encodedPayloadFromElements(tooLongFieldElements, maxPermissiblePayloadLength)
-		require.Error(t, err)
-	}
-}
-
-// TestTrailingNonZeros checks that any non-zero values that come after the end of the claimed payload length
-// cause an error to be returned.
-func TestTrailingNonZeros(t *testing.T) {
-	testRandom := random.NewTestRandom()
-	originalData := testRandom.Bytes(testRandom.Intn(1024) + 1)
-	encodedPayload := newEncodedPayload(NewPayload(originalData))
-
-	originalElements, err := encodedPayload.toFieldElements()
-	require.NoError(t, err)
-
-	fieldElements1 := make([]fr.Element, len(originalElements))
-	copy(fieldElements1, originalElements)
-
-	fieldElements2 := make([]fr.Element, len(originalElements))
-	copy(fieldElements2, originalElements)
-
-	// adding a 0 is fine
-	fieldElements1 = append(fieldElements1, fr.Element{})
-	_, err = encodedPayloadFromElements(fieldElements1, uint32(len(fieldElements1)*encoding.BYTES_PER_SYMBOL))
-	require.NoError(t, err)
-
-	// adding a non-0 is non-fine
-	fieldElements2 = append(fieldElements2, fr.Element{0, 0, 0, 1})
-	_, err = encodedPayloadFromElements(fieldElements2, uint32(len(fieldElements2)*encoding.BYTES_PER_SYMBOL))
-	require.Error(t, err)
-}
-
 // TestEncodeWithFewerElements tests that having fewer bytes than expected doesn't throw an error
 func TestEncodeWithFewerElements(t *testing.T) {
 	testRandom := random.NewTestRandom()
 	originalData := testRandom.Bytes(testRandom.Intn(1024) + 33)
+	encodedPayload := Payload(originalData).ToEncodedPayload()
 
-	encodedPayload := newEncodedPayload(NewPayload(originalData))
-
-	originalFieldElements, err := encodedPayload.toFieldElements()
+	originalBlob, err := encodedPayload.ToBlob(codecs.PolynomialFormCoeff)
 	require.NoError(t, err)
 
-	truncatedFieldElements := make([]fr.Element, len(originalFieldElements)-1)
-	// intentionally don't copy all the elements
-	copy(truncatedFieldElements, originalFieldElements[:len(originalFieldElements)-1])
+	truncatedCoefficients := make([]fr.Element, originalBlob.LenSymbols()-1)
+	copy(truncatedCoefficients, originalBlob.coeffPolynomial)
+	truncatedBlob := BlobFromCoefficients(truncatedCoefficients)
 
+	reconstructedEncodedPayload := truncatedBlob.ToEncodedPayload(codecs.PolynomialFormCoeff)
 	// even though the actual length will be less than the claimed length, we shouldn't see any error
-	reconstructedEncodedPayload, err := encodedPayloadFromElements(
-		originalFieldElements,
-		uint32(len(originalFieldElements))*encoding.BYTES_PER_SYMBOL)
-	require.NoError(t, err)
-	require.NotNil(t, reconstructedEncodedPayload)
+	require.Equal(t, encodedPayload, reconstructedEncodedPayload)
```

Comment:
What are you testing here? Here's how I'm understanding the test:
1. create encoded payload
2. turn it into a blob
3. truncate the blob elements
4. convert the blob with truncated elements back into an encoded payload
5. assert that the original encoded payload is equal to the `reconstructedEncodedPayload`

But I don't understand why these should be equal? Shouldn't the truncation have caused the `reconstructedEncodedPayload` to be different from the original?

Reply:
Agree with you and didn't want to spend too much energy trying to think if there was a specific reason why we needed this test, so just went ahead and deleted it: [084e95d](https://github.com/Layr-Labs/eigenda/pull/1841/commits/084e95d2460db012c27ab23546849a8611aaee98)

It was here only because I tried to keep it when I first refactored the library. But that likely changed the test's semantic and hence usefulness. Let me know if you think we should keep it.